Framework/Algorithms/src/SmoothData.cpp

```diff
@@ -92,17 +92,17 @@ void SmoothData::exec() {
 
     // Copy the X data over. Preserves data sharing if present in input
     // workspace.
-    outputWorkspace->setX(i, inputWorkspace->refX(i));
+    outputWorkspace->setSharedX(i, inputWorkspace->sharedX(i));
 
     // Now get references to the Y & E vectors in the input and output
     // workspaces
-    const MantidVec &Y = inputWorkspace->readY(i);
-    const MantidVec &E = inputWorkspace->readE(i);
-    MantidVec &newY = outputWorkspace->dataY(i);
-    MantidVec &newE = outputWorkspace->dataE(i);
+    const auto &Y = inputWorkspace->y(i);
+    const auto &E = inputWorkspace->e(i);
+    auto &newY = outputWorkspace->mutableY(i);
+    auto &newE = outputWorkspace->mutableE(i);
```

Comment:
Move these lines below the `if (npts == 0) {` --- otherwise you are paying potential overhead even if you just hit the `continue` in the `if` and don't actually use the references.